### ISSUE TITLE
Audio track not playing

### ISSUE THREAD

**Issue by IceLancerSR** [FoxAhead/Civ2-UI-Additions#7]
In game it is constantly showing track 12 but it isnt playing.
instal game, instal patch 1.3 us, add launcher to game, launch 

Pick music option doesnt do anything, i can choose different track but thats it, nothing happens.

![civ2](https://user-images.githubusercontent.com/4452102/65384007-2a997180-dd1d-11e9-8731-0b3b24793168.png)

**Comment by IceLancerSR** [comment]
P.S i turned off No CD disc check. And exe is not patched.

**Comment by IceLancerSR** [comment]
CD most definitely has files Track02.cda etc to Track12.cda

``

**Comment by FoxAhead** [comment]
How many optical drives are there in your system? Do you have virtual one? May be the reason is in drive order/letter?
I just can't imagine a reason why CD tracks doesn't work.
Do they work without this launcher? I mean if you just start the game without Civ2UIA is there CD music? May be at some point Windows 10 broke support of MCI API?
https://answers.microsoft.com/en-us/windows/forum/windows_10-performance-winpc/media-control-interface-nmci-not-working-properly/4931db24-a53a-46da-94db-7423e3d5540f

**Comment by IceLancerSR** [comment]
Only one virtual drive. I can see by using processactivity that game is accessing that drive and mcicida.dll when i activate music , but thats it
I ripped cd to with nero , so file is nrg , virtual drive is from poweriso , if i open iso with it , it shows track files and can even play them in it.

I cant enter game without mod , game crash. So i cant tell does music works without mod. 
I did manage to make it work with Music folder mp3 from other mod civ2patch

**Comment by IceLancerSR** [comment]
This is mixed image disc and it seems that all that i have tried , poweriso, ultra iso ,wincd emu cant mount such disc to be readable as cd audio? I tried mounting and playing it as audio with media player ,vlc player, none seems to be able to see audio part of cd.. I am out of clues what to try next

**Comment by FoxAhead** [comment]
Yes, I have such mixed mode CD image too downloaded somewhere from abandonware. 1st track - data, other - audio. I mounted it with Alcohol and tested both on physical Windows 7 and virtual Windows 10. In both systems such image plays without issue. In Win10 I just right click on disc and choose Play - default Media Player handle it. In Win7 again any media program can play it. For example Media Player Classic from K-Lite Pack can open it from menu File - Open Disc.

**Comment by IceLancerSR** [comment]
That doesnt work for me at all. I remember that quake and might and magic also didnt work.But they had solution:

https://github.com/hifi-unmaintained/ogg-winmm/
You can find a compiled build included with GoG CDDA games, it requires a folder named MUSIC in the game directory containing track01.ogg, track02.ogg etc. There's also a build available here.
THat is exactly what civ2patch is using

**Comment by FoxAhead** [comment]
Yes, I know about this feature from civ2patch. And actually there are several requests of this feature. But for now I'm just too busy (or may be lazy :) ) to implement it here. Furthermore I don't understand motivation. If CD audio-tracks are anyway converted into audio files then why just not use some background player (AIMP, Winamp etc.) and put there any music you want?

**Comment by IceLancerSR** [comment]
this is for an build arcade machine, there is no desktop

**Comment by FoxAhead** [comment]
Huh. This is a good point then! Can't promise anything but may be some day...
Let this issue be open reminding me or someone else who want to contribute to implement audio files playback from folder.

**Comment by IceLancerSR** [comment]
Ok i got audio to work from folder with original game , but it breaks with ur mod when u enter map ,probably ur player fixes take over

**Comment by FoxAhead** [comment]
Tested on my side on Win7. **Without** UIA it has the same problem that appeared in MCI after XP or Vista - tracks are not looping. It just plays one track and then stops. My patch aimed to fix that bug constantly polling CD status and at some conditions restarting next track. But with this WINMM-wrapper and **with** UIA it just stops playing at all. There is no crashes on my side but music plays only at OK-popup before the map. And on the map it mutes.
Also I noticed there is no drum sound at the main menu when you exit the map.

**Comment by scrutinizer80** [comment]
The old civ2patch had the option to play audio tracks from mp3/ogg. Sadly this feature wasn't implemented here and there doesn't seem to be a way of working with both in combination.
The only way to have a fully functioning Civ2 is currently still on a VM. despite the most impressive progress made on other fields. Not having the music just kills the experience.

**Comment by DARK-STAR03** [comment]
So currently, is the only way to run Civ2 fully functionally on a Win95/98 VM? Cause I've been trying to get it working on an XP VM and I've been having the same issue of the audio not working in-game, even with the patch.

**Comment by FoxAhead** [comment]
> So currently, is the only way to run Civ2 fully functionally on a Win95/98 VM? Cause I've been trying to get it working on an XP VM and I've been having the same issue of the audio not working in-game, even with the patch.

Civ2 MGE is usual Win32 application and with some steps it can run on Windows 7, 8, 10, 11 in usual way even without any compatibility settings.
1. Apply 64bit patch OR use CIV2UIA
2. For videos enable Intel Indeo Codec (https://forums.civfanatics.com/threads/indeo-codec-troubles-with-civ2-solved.362336/#post-14899055). And don't use noCD-patch - it disables videos!
3. If you are on Windows 10 - turn on legacy DirectPlay feature. Control Panel - Programs - Turn Windows Features on or off - Legacy Components - DirectPlay

Original Civ2 MGE CD-disc represents mixed mode cd. The first track is data and other are audio tracks. Either insert its physical copy in the CD-drive, or mount CD image with some software like Alcohol. If you are able to play music from such disc with some multimedia software then the game should handle it too. Also, again, remember not to use noCD-patch! In CIV2UIA turn off corresponding checkbox in the options. And make sure your CIV2.EXE is not hard-patched with some separate noCD-patch.

**Comment by DARK-STAR03** [comment]
> Civ2 MGE is usual Win32 application and with some steps it can run on Windows 7, 8, 10, 11 in usual way even without any compatibility settings.
> 
>     1. Apply 64bit patch OR use CIV2UIA
> 
>     2. For videos enable Intel Indeo Codec (https://forums.civfanatics.com/threads/indeo-codec-troubles-with-civ2-solved.362336/#post-14899055). And don't use noCD-patch - it disables videos!
> 
>     3. If you are on Windows 10 - turn on legacy DirectPlay feature. Control Panel - Programs - Turn Windows Features on or off - Legacy Components - DirectPlay
> 
> 
> Original Civ2 MGE CD-disc represents mixed mode cd. The first track is data and other are audio tracks. Either insert its physical copy in the CD-drive, or mount CD image with some software like Alcohol. If you are able to play music from such disc with some multimedia software then the game should handle it too. Also, again, remember not to use noCD-patch! In CIV2UIA turn off corresponding checkbox in the options. And make sure your CIV2.EXE is not hard-patched with some separate noCD-patch.

I have tried all of those, and no change, on both my XP VM and my host OS. (win11) The game runs fine, but no music once in game. (I also tried a version of the game that came pre-downloaded with your 64-bit Editbox Patcher, I was able to run it on my host OS without a disk image mounted but still no music.)

The only thing i can think of is that windows media player can't read the CD image of civ2MGE. Which is odd, cause it could read the non-MGE civ2 image just fine, and i got both files from the same site.

The disk images are in bin/cue format and the software I'm using to mount them is Virtual Clone Drive.
I've also tried the old civ2patch by vinceho. It kind of worked? The game was playing music but had other issues. On my XP VM i could not load save files, and on win11 i _could_ load save files, but i couldn't make a new save, as soon as i got to the point to input a name the textboxes glitched out and locked up the program, forcing me to close it via task manager.

The only thing i can think of that's the problem is my civ2 disk image, the mounting software, or if there's some way to make the two patches co-exist. I still have a lot to learn about messing with older games so my apologies if I've missed or misunderstood something, and thank you for your time and patience!
